File: Sean的leetcode/Leetcode/Add Strings.cpp
class Solution {
public:
    string addStrings(string num1, string num2) {
        int i = num1.size() - 1;
        int j = num2.size() - 1;
        string res;
        int carry = 0;
        while (i >= 0 || j >= 0 || carry) {
            int sum = 0;
            if (i >= 0) {
                sum += num1[i] - '0';
                i--;
            }
            if (j >= 0) {
                sum += num2[j] - '0';
                j--;
            }
            sum += carry;
            carry = sum / 10;
            sum %= 10;
            res += to_string(sum);
        }
        reverse(res.begin(), res.end());
        return res;
        
    }
};

File: Sean的leetcode/python/Add Strings.py
###
# String
# Time Complexity: O(n)
# Space Complexity: O(n)
###
class Solution(object):
    def addStrings(self, num1, num2):
        """
        :type num1: str
        :type num2: str
        :rtype: str
        """
        i = len(num1) - 1
        j = len(num2) - 1
        carry = 0
        re = []
        while i >= 0 or j >= 0 or carry:
            tmp = 0
            if i >= 0:
                tmp += int(num1[i])
                i -= 1
            if j >= 0:
                tmp += int(num2[j])
                j -= 1
            tmp += carry
            re.append(str(tmp % 10))
            carry = tmp / 10
        return ''.join(re[::-1])
        